From 73625e20eaf6a41d5c39bc667c67f9e7fdaebdfe Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Mon, 22 Jul 2019 23:14:41 -0400
Subject: [PATCH] aufs5: fix build on v5.3+

commit 9af0f1a46bbb6ad9ee8b35957251f4aa826b023f changes the rw_sem
task owner to an atomic type. We switch to using the introduced
helper function to get aufs compiling against 5.3

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/aufs/i_op.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/aufs/i_op.c b/fs/aufs/i_op.c
index 4f5001ebd0f4..3455f2ecb407 100644
--- a/fs/aufs/i_op.c
+++ b/fs/aufs/i_op.c
@@ -641,7 +641,7 @@ int au_pin_hdir_relock(struct au_pin *p)
 static void au_pin_hdir_set_owner(struct au_pin *p, struct task_struct *task)
 {
 #if !defined(CONFIG_RWSEM_GENERIC_SPINLOCK) && defined(CONFIG_RWSEM_SPIN_ON_OWNER)
-	p->hdir->hi_inode->i_rwsem.owner = task;
+	atomic_long_set(&p->hdir->hi_inode->i_rwsem.owner, (long)task);
 #endif
 }
 
-- 
2.19.1

